Source/Core/CylinderGeometry.js

```diff
@@ -203,6 +200,10 @@ define([
         var vertexFormat = cylinderGeometry._vertexFormat;
         var slices = cylinderGeometry._slices;
 
+        if ((length <= 0) || ((topRadius <= 0) && (bottomRadius <= 0))) {
```

Comment:
Small logic error here.  We want to throw if both top and bottom radius is zero, or if either top or bottom radius is less than zero.  Right now the cylinder will render if  topRadius = -10, bottomRadius = 10